openlibrary/plugins/openlibrary/home.py

```diff
@@ -124,12 +130,16 @@ def generic_carousel(query=None, subject=None, work_id=None, _type=None,
     return storify(books) if books else books
 
 @public
-def readonline_carousel():
+def readonline_carousel(query=None):
     """Return template code for books pulled from search engine.
        TODO: If problems, use stock list.
     """
     try:
-        data = random_ebooks()
+        if query in CAROUSELS_PRESETS:
+            query = CAROUSELS_PRESETS[query]
+            data = search_ebooks(query)
+        else:
+            data = random_ebooks()
```

Comment:
We likely want to treat it as a normal query if it's not a preset.

```suggestion
        if query in CAROUSELS_PRESETS:
            query = CAROUSELS_PRESETS[query]
        data = search_ebooks(query) if query else random_ebooks()
```